controllers/secret_controller_test.go

```diff
@@ -0,0 +1,185 @@
+package controllers
+
+import (
+	"context"
+	"reflect"
+	"testing"
+
+	api "github.com/k8ssandra/k8ssandra-operator/api/v1alpha1"
+	"github.com/k8ssandra/k8ssandra-operator/test/framework"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
```

Comment:
It would be good to see another test that covers removing a secret. Let's say I have clusters A, B, and C with a secret replicated to all 3. Now I want  remove the secret from C but not A and B.

Reply:
I don't know what the process with deletion would be. Currently it's just replicating, but if you disconnect replication - does that indicate one wants to delete them?

Reply:
I feel like there is a broader design discussion here to be had.

Reply:
I thought about this some more. Yes, I do think the replicated secrets should be deleted. They are basically child components of the replicated secrets. 

I do not think that the source secret that the user provides should be deleted though.